cmd/gardenlet/app/migration.go

```diff
@@ -106,3 +118,23 @@ func migrateDeprecatedTopologyLabels(ctx context.Context, log logr.Logger, seedC
 
 	return flow.Parallel(taskFns...)(ctx)
 }
+
+// TODO(Wieneo): Remove this function when feature gate RemoveAPIServerProxyLegacyPort is removed
+func migrateRemoveAPIServerProxyLegacyPortFeatureGate(ctx context.Context, log logr.Logger, gardenClient client.Client, seedName string) error {
+	shootList := &gardencorev1beta1.ShootList{}
+	if err := gardenClient.List(ctx, shootList); err != nil {
+		return err
+	}
+
+	for _, k := range shootList.Items {
+		if specSeedName, statusSeedName := gardenerutils.GetShootSeedNames(&k); gardenerutils.GetResponsibleSeedName(specSeedName, statusSeedName) != seedName {
+			continue
+		}
+
+		if cond := v1beta1helper.GetCondition(k.Status.Constraints, gardencorev1beta1.ShootAPIServerProxyUsesHTTPProxy); cond == nil || cond.Status != gardencorev1beta1.ConditionTrue {
+			return errors.New("all shoots on this seed need to have APIServerProxyUsesHTTPProxy constraint set to true")
```

Comment:
Can you make this error message more helpful for operators that are not in context of GEP-30?
For example:
```
the `proxy` port on the Istio ingress gateway cannot be removed until all API server proxies in all Shoots on this Seed have been reconfigured to use the `tls-tunnel` port instead, i.e., the `RemoveAPIServerProxyLegacyPort` feature gate can only be enabled once all Shoots have the `APIServerProxyUsesHTTPProxy` constraint with status `True`
```